File: backend/src/main/java/edu/universidad/centromedico/model/Catalogos.java
package edu.universidad.centromedico.model;

import java.util.Set;

/**
 * Catálogos cerrados de carreras y especialidades. Fuente de verdad para validar
 * que el admin no registre valores basura (p. ej. carrera "asd"). Debe mantenerse
 * en sintonía con `frontend/src/constants/catalogos.js`.
 */
public final class Catalogos {

    private Catalogos() {}

    public static final Set<String> CARRERAS = Set.of(
        "Ingenieria de Sistemas", "Ingenieria Industrial", "Ingenieria Civil",
        "Ingenieria Electronica", "Ingenieria Mecanica", "Ingenieria Quimica",
        "Ingenieria Economica", "Arquitectura", "Ciencias"
    );

    public static final Set<String> ESPECIALIDADES = Set.of(
        "Medicina General", "Cardiologia", "Dermatologia", "Endocrinologia",
        "Ginecologia", "Neurologia", "Odontologia", "Oftalmologia", "Pediatria",
        "Psicologia", "Radiologia", "Traumatologia"
    );

    public static final Set<String> TIPOS_SANGRE = Set.of(
        "O+", "O-", "A+", "A-", "B+", "B-", "AB+", "AB-"
    );

    /** El tipo de sangre es opcional; si viene, debe ser uno de la lista. */
    public static void validarTipoSangre(String tipoSangre) {
        if (tipoSangre != null && !tipoSangre.isBlank() && !TIPOS_SANGRE.contains(tipoSangre.trim())) {
            throw new RuntimeException("El tipo de sangre no es válido");
        }
    }

    public static void validarCarrera(String carrera) {
        if (carrera == null || !CARRERAS.contains(carrera.trim())) {
            throw new RuntimeException("La carrera no es válida; selecciona una de la lista");
        }
    }

    public static void validarEspecialidad(String especialidad) {
        if (especialidad == null || !ESPECIALIDADES.contains(especialidad.trim())) {
            throw new RuntimeException("La especialidad no es válida; selecciona una de la lista");
        }
    }
}

File: frontend/src/constants/catalogos.js
/**
 * Catálogos compartidos (carreras y especialidades). Fuente única para los
 * combos del frontend; el backend valida contra las mismas listas (ver
 * `model/Catalogos.java`). Mantener ambas en sintonía.
 */

// Carreras de la UNI.
export const CARRERAS = [
  'Ingenieria de Sistemas', 'Ingenieria Industrial', 'Ingenieria Civil',
  'Ingenieria Electronica', 'Ingenieria Mecanica', 'Ingenieria Quimica',
  'Ingenieria Economica', 'Arquitectura', 'Ciencias',
];

// Especialidades médicas (incluye las sembradas en init_db.sql).
export const ESPECIALIDADES = [
  'Medicina General', 'Cardiologia', 'Dermatologia', 'Endocrinologia',
  'Ginecologia', 'Neurologia', 'Odontologia', 'Oftalmologia', 'Pediatria',
  'Psicologia', 'Radiologia', 'Traumatologia',
];

// Grupos sanguíneos (debe coincidir con Catalogos.TIPOS_SANGRE en el backend).
export const TIPOS_SANGRE = ['O+', 'O-', 'A+', 'A-', 'B+', 'B-', 'AB+', 'AB-'];
